## 문제 1 : 혼동행렬 파이썬 코드 작성

In [1]:
tp = 90
tn = 45
fp = 10
fn = 35

total = tp + tn + fp + fn

accuracy = (tp + tn) / total
precision = tp / (tp + fp)
recall = tp / (tp + fn)
f1_score = 2 * (precision * recall) / (precision + recall)

print("TP:", tp)
print("TN:", tn)
print("FP:", fp)
print("FN:", fn)

print(f"Accuracy : {accuracy:.2f}")
print(f"Precision: {precision:.2f}")
print(f"Recall   : {recall:.2f}")
print(f"F1-score : {f1_score:.2f}")

TP: 90
TN: 45
FP: 10
FN: 35
Accuracy : 0.75
Precision: 0.90
Recall   : 0.72
F1-score : 0.80


## 문제 2. 결과 해석해보기
**답안지 (각 질문별로 아래 답안지 작성 예시처럼 답을 추가하시고 제출해주시면 됩니다. 코드 셀, 마크다운 모두 상관없음)**
1. 도출된 정확도 수치는 무엇을 의미하는가?
2. 모델이 질병이라고 예측한 사람은 총 몇 명인가?
3. 도출된 정밀도 수치는 무엇을 의미하는가?
4. 실제 질병 환자는 총 몇 명인가?
5. 도출된 재현율 수치는 무엇을 의미하는가?
6. 이 모델이 놓친 실제 환자는 몇 명인가?
7. 질병 환자를 놓치지 않는 것이 중요하다면 어떤 지표를 우선 확인해야 하는가?

1.  **도출된 정확도 수치는 무엇을 의미하는가?**
    전체 180명 중 75%인 135명(90+45)을 올바르게(질병/정상) 예측했다는 의미

2.  **모델이 질병이라고 예측한 사람은 총 몇 명인가?**
    모델이 질병이라고 예측한 사람은 TP 90명과 FP 10명을 합한 100명이다.

3.  **도출된 정밀도 수치는 무엇을 의미하는가?**
    모델이 '질병'이라고 예측한 100명 중에서 실제로 질병인 사람이 90명(90%)이라는 의미이다. 모델이 질병이라고 판단했을 때 그 예측이 얼마나 정확한지를 나타낸다.

4.  **실제 질병 환자는 총 몇 명인가?**
    실제 질병 환자는 TP 90명과 FN 35명을 합한 125명이다.

5.  **도출된 재현율 수치는 무엇을 의미하는가?**
    실제 질병 환자 125명 중에서, 모델이 질병이 있다고 찾아낸 비율이 72%라는 의미이다.


6.  **이 모델이 놓친 실제 환자는 몇 명인가?**
    이 모델이 놓친 실제 환자는 FN 값인 35명이다.

7.  **질병 환자를 놓치지 않는 것이 중요하다면 어떤 지표를 우선 확인해야 하는가?**
 재현율(Recall)을 우선적으로 확인해야 한다. 재현율은 실제 질병 환자 중에서 모델이 올바르게 질병이라고 예측한 비율을 나타내기 때문이다..